shaders/src/main/glsl/samples/310es/binarysearch_bw.frag

```diff
@@ -0,0 +1,141 @@
+#version 310 es
+
+/*
+ * Copyright 2019 The GraphicsFuzz Project Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+precision highp float;
+
+layout(location = 0) out vec4 _GLF_color;
+
+uniform vec2 injectionSwitch;
+
+uniform vec2 resolution;
+
+struct Obj{
+    int prime_numbers[10];
+};
+
+vec2 brick(vec2 uv) {
+    int a = 4;
+    do {
+        uv.y -= step(injectionSwitch.y, uv.x);
+        a--;
+    } while (a > int(injectionSwitch.x));
+    int b = 3;
+    do {
+        uv.y -= step(injectionSwitch.y, uv.x) + float(a);
+        b--;
+    } while (b > int(injectionSwitch.x));
+    int c = 2;
+    do {
+        uv.y -= step(injectionSwitch.y, uv.x) + float(a) + float(b);
+        c--;
+    } while (c > int(injectionSwitch.x));
+    int d = 1;
+    do {
+        uv.y -= step(injectionSwitch.y, uv.x) + float(a) + float(b) + float(c);
+        d--;
+    } while (d > int(injectionSwitch.x));
+    return fract(uv);
+}
+
+float patternize(vec2 uv) {
+    vec2 size = vec2(0.45);
+    vec2 st = smoothstep(size, size, uv);
+     switch (int(mod(gl_FragCoord.y, 5.0))) {
+        case 0:
+            return mix(pow(st.x, injectionSwitch.y), st.x, size.y);
+        break;
+        case 1:
+            return mix(pow(uv.y, injectionSwitch.y), st.y, size.x);
+        break;
+        case 2:
+            discard;
+        break;
+        case 3:
+            return mix(pow(uv.y, injectionSwitch.y), uv.y, size.y);
+        break;
+        case 4:
+            return mix(pow(st.y, injectionSwitch.y), st.x, size.x); 
+        break;
+     }
+}
+
+int binarySearch(Obj obj, int x) {
+    int l = 0, r = 9;
+    while (l <= r) {
+        int m = (l + r) / 2;
+        if (obj.prime_numbers[m] == x) {
+            return m;
+        }
+        
+        if (obj.prime_numbers[m] < x) {
+            l = m + 1;  
+        } else {
+            r = m - 1; 
+        }
+    }
+    return -1;
+}  
+
+void main() {
+    Obj obj;
+    for (int i = 0; i < 10; i++) {
+        int x;
```

Comment:
Is this an unused variable? This is fine I guess, unless glslangValidator complains. Did you run glslangValidator on all shaders?

Reply:
Oops! it's indeed unused variable. I did run with the glslangValidator but it didn't give me any warning message. Thanks for pointing out.